src/download_gcs_file.py

```diff
@@ -0,0 +1,172 @@
+import argparse
+import concurrent.futures
+from pathlib import Path
+
+import pandas as pd
+from boto3.session import Session
+from google.cloud import storage
+from loguru import logger
+
+max_workers = 5
+GCS_PROJECT = "ida-prd"
+
+
+def create_parser() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="download creative from csv.")
+    parser.add_argument(
+        "csv_path",
+        type=str,
+        help="csv.gz file path.",
+    )
+    parser.add_argument(
+        "save_dir",
+        type=str,
+        help="creative directory.",
+    )
+
+    parser.add_argument(
+        "--negocia_profile",
+        type=str,
+        default="default",
+        help="aws negocia profile name.",
+    )
+
+    parser.add_argument(
+        "--negocia_s3_uri",
+        type=str,
+        default=None,
+        help="negocia s3 creative dir",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def get_s3_client(profile_name):
+    """
+    指定されたプロファイル名を使用してS3クライアントを取得します。
+
+    Args:
+        profile_name (str): AWSプロファイル名。
+
+    Returns:
+        boto3.client: S3クライアントオブジェクト。
+    """
+    session = Session(profile_name=profile_name)
+    client = session.client("s3")
+    return client
+
+
+def _list_all_objects(s3_client, bucket, prefix, **kwargs):
+    """
+    指定されたバケットとプレフィックスに一致する全てのオブジェクトをリストします。
+
+    Args:
+        s3_client (boto3.client): S3クライアントオブジェクト。
+        bucket (str): バケット名。
+        prefix (str): オブジェクトのプレフィックス。
+        **kwargs: その他のオプション引数。
+
+    Returns:
+        list: 全てのオブジェクトを含むレスポンスのリスト。
+    """
+    kwargs.update(Bucket=bucket, Prefix=prefix)
+
+    def _list_objects(continuation_token=None):
+        if continuation_token is None:
+            response = s3_client.list_objects_v2(**kwargs)
+        else:
+            response = s3_client.list_objects_v2(
+                ContinuationToken=continuation_token, **kwargs
+            )
+        return response
+
+    responses = []
+    response = _list_objects()
+    responses.append(response)
+    while response["IsTruncated"]:
+        continuation_token = response["NextContinuationToken"]
+        logger.info(f"{continuation_token=}")
+        response = _list_objects(continuation_token)
+        responses.append(response)
+    return responses
+
+
+def _get_negocai_s3_file_list(negocia_profile, negocia_s3_uri):
```

Comment:
- negocai -> negocia
```suggestion
def _get_negocia_s3_file_list(negocia_profile, negocia_s3_uri):
```